src/ai/defaultai.cc

```diff
@@ -5664,13 +5663,13 @@ BuildingNecessity DefaultAI::check_building_necessity(BuildingObserver& bo,
 			inputs[22] = (basic_economy_established) ? -1 : 1;
 			inputs[23] = (msites_in_constr() > 0) ? 1 : -2;
 			inputs[24] = (msites_in_constr() > 1) ? 1 : -2;
-			inputs[25] = (wood_policy_ != WoodPolicy::kAllowRangers) * 1;
+			inputs[25] = (wood_policy_ != WoodPolicy::kAllowRangers) * -1;
 			if (gametime > Time(90 * 60)) {
```

Comment:
Is this // late game, afte 90 minutes ?

Reply:
exactly

Reply:
So maybe we should heva some constants like kMidGame, kLateGame ?

Reply:
there are much more points in time, which are evaluated  in the AI which would end up with a lot of constants then.

Reply:
on a close look the milliseconds were missing so I added the 1000 in all instances I found.